### ISSUE TITLE
Can no longer push my commits

### ISSUE THREAD

**Issue by MiTribunskiy** [desktop/desktop#6868]
Good day.

## Description
When I change files in the local repository and try to commit the changes to the origin remote, I get an error "Authentication failed".
![error](https://user-images.githubusercontent.com/46776781/52910997-0d488480-32b0-11e9-96f9-b9f53eabab1c.png)

## Version

* GitHub Desktop: 1.6.2

* Operating system: Windows 7 SP1

## Steps to Reproduce

1. `make local changes to the repository's files`
2. `in GitHub Desktop click on "Commit to master"` 
![commit to master 2](https://user-images.githubusercontent.com/46776781/52911230-163a5580-32b2-11e9-9753-c876893fa43c.png)
3. `click on "Push origin" when GitHub Desktop shows this menu:` 
![push origin](https://user-images.githubusercontent.com/46776781/52911136-d2931c00-32b0-11e9-9e6f-e2230f48c4f7.png)

### Expected Behavior

Changes were commited to the origin remote.

### Actual Behavior

No changes were transfered. Error "Authentication failed" is shown.
![error again](https://user-images.githubusercontent.com/46776781/52911147-f5253500-32b0-11e9-9f46-53f87c759974.png)

## Additional Information

I still can pull successfully via GitHub Desktop from the origin remote (if I change some files in the remote repository on github.com via browser).
When I try to push using the command line (like this: https://help.github.com/enterprise/2.16/user/articles/adding-a-file-to-a-repository-using-the-command-line),  commits are passed successfully. 
Approximately 2 weeks ago I also tested pulls and pushes via GitHub Desktop, and they worked fine.
Repositories were cloned with HTTPS (currently testing with tutorial repository):
![https](https://user-images.githubusercontent.com/46776781/52911150-02422400-32b1-11e9-8f56-5f5a031555ed.png)

I am logged in:
![logged](https://user-images.githubusercontent.com/46776781/52911151-08380500-32b1-11e9-97ca-eaaf34bd6e48.png)
Repositories are public and not archived.
I also tried to: 
1. relog in Desktop (both by entering the password and via browser)
2. relog in browser
3. revoke authorization and give it back in my profile's settings (Settings > Applications > Authorized OAuth Apps)
4. create new repository and push changes (without success)

### Logs

Only these lines appear in the log file when I click on "Commit to master" or "Push origin":

2019-02-17T08:24:31.286Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: hello-world - MiTribunskiy (has token)
2019-02-17T08:24:32.557Z - info: [ui] Executing push: git -c credential.helper= push origin master:master --progress (took 1.251s)
2019-02-17T08:26:00.527Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: hello-world - MiTribunskiy (has token)
2019-02-17T08:26:01.788Z - info: [ui] Executing push: git -c credential.helper= push origin master:master --progress (took 1.237s)

**Comment by Fedefreez** [comment]
Does this error pop up since you installed GitHub desktop or before that bug you could push?

**Comment by MiTribunskiy** [comment]
@Fedefreez Hello. Yes, around 2 weeks ago I tested push and pull options via GitHub Desktop, and they worked. 
Yes, I have 2-factor auth. I also had it when I tested pushes 2 weeks ago.

I've just disabled it and relogged. 
Unfortunately, when I try to push, I still get the same error "Authentication failed".

These lines appeared in the log file:
2019-02-17T15:52:45.068Z - info: [ui] [AppStore] loading 4 repositories from store
2019-02-17T15:52:45.069Z - info: [ui] [AppStore] found account: MiTribunskiy (MiTribunskiy)
2019-02-17T15:52:45.315Z - info: [ui] launching: 1.6.2 (Windows 6.1.7601)
2019-02-17T15:52:45.315Z - info: [ui] execPath: 'C:\Users\ _Username_ \AppData\Local\GitHubDesktop\app-1.6.2\GitHubDesktop.exe'
2019-02-17T15:52:52.333Z - info: [ui] [AppStore] removing account MiTribunskiy (MiTribunskiy) from store
2019-02-17T15:54:10.678Z - info: [ui] [SignInStore] initializing OAuth flow
2019-02-17T15:54:10.706Z - info: [main] opening in browser: https://github.com/login/oauth/authorize?client_id= _authorisation link_
2019-02-17T15:54:14.668Z - info: [main] Received possible protocol arguments: 3
2019-02-17T15:54:14.668Z - info: [main] Processing protocol url
2019-02-17T15:54:14.670Z - info: [ui] [Dispatcher] requesting authenticated user
2019-02-17T15:54:15.569Z - info: [ui] [SignInStore] account resolved
2019-02-17T15:54:15.571Z - info: [ui] [AppStore] adding account MiTribunskiy (MiTribunskiy) to store
2019-02-17T15:54:24.006Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: hello-world - MiTribunskiy (has token)
2019-02-17T15:54:25.275Z - info: [ui] Executing push: git -c credential.helper= push origin master:master --progress (took 1.237s)
2019-02-17T15:54:29.510Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: hello-world - MiTribunskiy (has token)
2019-02-17T15:54:30.793Z - info: [ui] Executing push: git -c credential.helper= push origin master:master --progress (took 1.250s)
2019-02-17T15:54:48.215Z - info: [ui] Background fetch for 4 repositories took 2.651sec
2019-02-17T15:56:29.152Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: hello-world - MiTribunskiy (has token)
2019-02-17T15:56:30.385Z - info: [ui] Executing push: git -c credential.helper= push origin master:master --progress (took 1.205s)

**Comment by Fedefreez** [comment]
You said that if you push via cmd line it works right?
I think the cause might be some corrupted file.
Try to uninstall GH desktop from the control panel and delete its files from:
%appdata%
%localappdata%
PREFETCH
(just copy & paste them in Cortana and the folder will pop out)

Then install it (and keep 2F auth disabled, if it doesn't still work then you can enable it again).

Some more things:
are you the owner of that repo or a collaborator?
Does the error show up also with other repos or only with the one you're using?

**Comment by MiTribunskiy** [comment]
@Fedefreez  Yes, I've just tried to push via cmd again and it worked (I pushed via Git Bash, is it ok?).

I disabled 2F auth and deleted GitHub Desktop from the control panel and then from %appdata% and
%localappdata%. When I reinstalled and logged in, I still couldn't push. Pulls are working.

When I tried to create a local repo and publish it, information about the repo appeared on github.com:

![repo1](https://user-images.githubusercontent.com/46776781/52935928-cd46d780-336b-11e9-9ac2-c0fcc68433c5.PNG)


However, the repo was empty (local repo has files):
![repo2](https://user-images.githubusercontent.com/46776781/52935933-d172f500-336b-11e9-99a9-91f81e2fabc0.PNG)

I am the owner of the repo and don't have collaborators. I can't push neither to the repos I created, nor to the ones I forked.

**Comment by Fedefreez** [comment]
What command do you use to push via git bash?

**Comment by MiTribunskiy** [comment]
To push via Git Bush I open Git Bush in the repo folder and enter:
1. `git add .`
2. `git commit -m test.txt` or any other file
3. `git push origin master`

Repo's settings in GH Desktop:
![repo_settings](https://user-images.githubusercontent.com/46776781/52964916-a06ae280-33b4-11e9-8fd8-89e45e27b2e4.PNG)
![repo_settings_2](https://user-images.githubusercontent.com/46776781/52964922-a365d300-33b4-11e9-8991-176f5cb4f5ac.PNG)

Repo's settings at github.com:
![repo_settings_3](https://user-images.githubusercontent.com/46776781/52964938-ab257780-33b4-11e9-98e2-715d77331639.PNG)
![repo_settings_4](https://user-images.githubusercontent.com/46776781/52964944-acef3b00-33b4-11e9-87e8-277184c27114.PNG)
![repo_settings_5](https://user-images.githubusercontent.com/46776781/52964950-af519500-33b4-11e9-8798-563e0739adea.PNG)

No, I don't use VPN.

**Comment by Fedefreez** [comment]
Ok then let's try to execute the commands that GH desktop executes and see if we'll get a bit more details.
Try in git bash (from a test repo folder):
git -c credential.helper= push origin master:master
git -c credential.helper= push origin master --set-upstream --progress

**Comment by MiTribunskiy** [comment]
I deleted 1 file, edited a 2nd one and created a 3rd one.
Then I entered: 
`git -c credential.helper= push origin master:master`

Then I did the same steps with this command: 
`git -c credential.helper= push origin master --set-upstream --progress`

This is what I got in cmd:
![gitbush](https://user-images.githubusercontent.com/46776781/52972686-b2587f80-33cc-11e9-88fa-34ef610bead8.PNG)

Although it says that `Everything up-to-date`, the local repo and the remote one are not synched. The remote repo at github.com was not affected by the commands.

**Comment by Fedefreez** [comment]
You need to commit the edited file first :D

**Comment by MiTribunskiy** [comment]
Oops, I apologise.

I edited file again, commited it and used `git -c credential.helper= push origin master:master`
Then I did the same with `git -c credential.helper= push origin master --set-upstream --progress`

Both worked! The remote repo was changed by both of them: https://github.com/MiTribunskiy/hello-world/blob/master/test.txt

This is what I got in cmd:
![gitbush2](https://user-images.githubusercontent.com/46776781/52974279-044fd400-33d2-11e9-9877-fc673ad2311a.PNG)

**Comment by Fedefreez** [comment]
w... what?
The log says everything is fine.
Executing push as the GH desktop program does works, but it throws an error.
Why does that even happen? I mean, you executed the same cmds... the only thing I can think
of is that the credentials used by the program are corrupted/wrong for some reason.

So, With this I'll need you to go in GH desktop > options > git > send screenshot.

**Comment by MiTribunskiy** [comment]
I deleted GH Desktop, deleted its files from %appdata% and %localappdata% , downloaded new installer, reinstalled - still Authentification error when I try to push.

This screenshot?
![options_git](https://user-images.githubusercontent.com/46776781/52977376-10da2980-33de-11e9-8564-b13f22a0f119.PNG)

**Comment by Fedefreez** [comment]
I think that now we should fragment the problem and exclude things to get as colse as possible to the problem itself. Now, as first step try to make another accout (your can use  10minutesmail.com for the email) and log in GH desktop. After tell me if it works. I think that the problem might be sticked to your account for some reason.

**Comment by o12800915** [comment]
This is another acc. Still the same error

I created 1 repository on my computer and 1 on github.com. 
I could pull the repo from github, but I couldn't publish the repo I created on my computer. Then when I edited a file from the repo I pulled from github.com and tried to push it back I also got authentification error.
Should I add this acc as a collaborator?

**Comment by Fedefreez** [comment]
No, if it happens also with that acc, can you try to install GH desktop in another pc or on a VM? Maybe it's just your pc...
If it works on another pc we'll just copy the config. On the 2nd pc use your main acc, se secondary account can be deleted.

**Comment by shiftkey** [comment]
@MiTribunskiy there's something interesting in the logs that I wanted to call out:

```
2019-02-17T15:54:15.571Z - info: [ui] [AppStore] adding account MiTribunskiy (MiTribunskiy) to store
2019-02-17T15:54:24.006Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: hello-world - MiTribunskiy (has token)
2019-02-17T15:54:25.275Z - info: [ui] Executing push: git -c credential.helper= push origin master:master --progress (took 1.237s)
2019-02-17T15:54:29.510Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: hello-world - MiTribunskiy (has token)
2019-02-17T15:54:30.793Z - info: [ui] Executing push: git -c credential.helper= push origin master:master --progress (took 1.250s)
2019-02-17T15:54:48.215Z - info: [ui] Background fetch for 4 repositories took 2.651sec
2019-02-17T15:56:29.152Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: hello-world - MiTribunskiy (has token)
2019-02-17T15:56:30.385Z - info: [ui] Executing push: git -c credential.helper= push origin master:master --progress (took 1.205s)
```

It looks like these `git push` operations are completing without error, so I'm curious why the changes aren't ending up where you expect them to be.

Are you able to find the original errors that occurred with that screenshot in the log file? There should be some underlying `git` messages that might help us with that context..

**Comment by Fedefreez** [comment]
Yea I was asking myself the reason of the missing errors in the logs too...

**Comment by shiftkey** [comment]
Marking this as `more-information-needed` because I haven't yet seen the log containing the original `git push` error and the underlying output from Git related to it.

![](https://user-images.githubusercontent.com/46776781/52911147-f5253500-32b0-11e9-9f46-53f87c759974.png)

There's a lot of extra troubleshooting in here, which is great, but I'd love to start from "what happened here?" because that will help us investigate the original problem...

**Comment by Fedefreez** [comment]
Yes, I was doing the only thing I could think of to spot the problem because I thought that the log MiT sent was the original one: maybe he's wrong but presuming he is right I tried to get as close to the problem as possible. (we're still far tho)

**Comment by MiTribunskiy** [comment]
@shiftkey The problem is there is no message about the authentification error in the log file. I've just tried to push again, got error:
![error_3](https://user-images.githubusercontent.com/46776781/53037669-d7f88e00-348b-11e9-8605-3a82bd3efcf5.png)

My time is UTC +3

And this is the log file:
[2019-02-19.desktop.production.log](https://github.com/desktop/desktop/files/2881410/2019-02-19.desktop.production.log)
I redacted only _Username_ at line 4 and _authentification_link_ at line 12

@Fedefreez I don't have another computer right now, so I'll try to install VM. I'm sorry, is there VM you would recommend?

**Comment by Fedefreez** [comment]
Virtualbox is ok, you also need windows' iso image, but wait, let's first try to solve this as @shiftkey says. Maybe he knows a better solution than mine.

**Comment by MiTribunskiy** [comment]
Just to break the silence: I understand that it is hard to reproduce this error considering there is no information in the log file about it. So it is hard to help me fix this issue.
At least I can pull. I'll push with Git Bash when needed. 

I'll install Virtualbox on the coming weekends and try to push again. 
If I fix this problem somehow, I'll write about solution here.

**Comment by steveward** [comment]
@MiTribunskiy just to make sure I understand the issue -- you are able to successfully clone and pull from repositories in your GitHub account, but you are unable to push any changes from within GitHub Desktop. Is that correct?

Have you tried removing the token GitHub Desktop creates when you log in to the application? You can do this by going to `Credential Manager` > `Windows Credentials` and deleting the `GitHub - https://api.github.com/MiTribunskiy` entry. Then try logging out and logging back in to GitHub Desktop, and let us know if that gets things working for you.

**Comment by MiTribunskiy** [comment]
@steveward Yes, that is correct.

Unfortunately, it didn't help.
I deleted `GitHub - https://api.github.com/MiTribunskiy` entry from `Windows Credentials`, signed out,  deleted GH Desktop, deleted its files from %appdata% and %localappdata%, revoked authorization for GH Desktop in the settings of my account at github.com, downloaded new installer from https://desktop.github.com/, installed, logged in. Then I successfully pulled repository, changed its file, commited it but still got an Authentification error when I tried to push.

This is the log file:
[2019-02-20.desktop.production.log](https://github.com/desktop/desktop/files/2886289/2019-02-20.desktop.production.log)

**Comment by shiftkey** [comment]
@MiTribunskiy the thing that I'm trying to wrap my head around is that, based on the logs you just sent through, that no error was reported from the app:

```
2019-02-20T20:00:16.589Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: hello-world - MiTribunskiy (has token)
2019-02-20T20:00:18.372Z - info: [ui] Executing push: git -c credential.helper= push origin master:master --progress (took 1.775s)
...
2019-02-20T20:01:40.072Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: hello-world - MiTribunskiy (has token)
2019-02-20T20:01:41.259Z - info: [ui] Executing push: git -c credential.helper= push origin master:master --progress (took 1.154s)
```

What I was hoping to see was something like this (this came from #6859):

```
2019-02-15T09:19:16.468Z - error: [ui] `git -c credential.helper= clone --recursive --progress -- [url] [path]` exited with an unexpected code: 128.
Cloning into [path]'...
remote: Invalid username or password.
fatal: Authentication failed for '[url]'

(The error was parsed as 3: Authentication failed. You may not have permission to access the repository or the repository may have been archived. Open options and verify that you're signed in with an account that has permission to access this repository.)
```

This is the code that generates this context:

https://github.com/desktop/desktop/blob/3223b072978890b0be9fc043a9101b0ae1b63713/app/src/lib/git/core.ts#L152-L169

Can you share what's in your `~/.gitconfig` global config? Feel free to hide any identifying information - I'm looking for config that might be authentication related.

For example, this is how you'd force all Git operations on the machine to use SSH, which would affect Desktop in this case:

```
# Enforce SSH
[url "ssh://git@github.com/"]
  insteadOf = https://github.com/
```

I'm still puzzled why it's not surfacing the underlying Git error, but let's try and eliminate some other environmental issues first.

**Comment by MiTribunskiy** [comment]
@shiftkey Is it the correct one?
[gitconfig.txt](https://github.com/desktop/desktop/files/2886478/gitconfig.txt)
I got it with `git config --global --edit` command in Git Bash.

It was named as .gitconfig but I couldn't attach it to the comment (unsupported extension), so I renamed it and saved as .txt.
I redacted _Username_ at line 2

**Comment by Fedefreez** [comment]
At this point you'd need to see the win crdential content, if the error is originated by that file: I don't think it's that thi because he resetted it and using credential.helper in git bash works.
P.S. @shiftkey when you find some free time can you check out my issue? Thx

**Comment by steveward** [comment]
@MiTribunskiy are you still unable to push changes successfully using GitHub Desktop?

**Comment by MiTribunskiy** [comment]
@steveward Hello. I have just tried to push and it worked! 

I pushed the same commits I couldn't push 3 weeks ago.
I can also push commits I have just created.

No idea what fixed this problem -  I did not install or delete anything (beside regular Windows Updates).

Log file for 2 successful pushes:
[2019-03-14.desktop.production.log](https://github.com/desktop/desktop/files/2968431/2019-03-14.desktop.production.log)
